src/scipp/core/operations.py

```diff
@@ -380,7 +380,16 @@ def label_based_index_to_positional_index(
         coord,
         index,
     )
-    return (dim, inds[0] if len(inds) == 1 else slice(*inds))
+    if len(inds) == 1:
```

Comment:
Why this `if` statement?

Reply:
If `index` was a slice then `len(inds) > 1` but if `index` was a variable then `len(inds) == 1`. So this if statement checks which case we are in. If we were passed a slice the out of range check is not necessary because in that case we just return an empty slice.